mlrun/model.py

```diff
@@ -377,6 +378,115 @@ def source(self, source):
 
         self._source = source
 
+    def build_config(
+        self,
+        image="",
+        base_image=None,
+        commands: list = None,
+        secret=None,
+        source=None,
+        extra=None,
+        load_source_on_run=None,
+        with_mlrun=None,
+        auto_build=None,
+        requirements=None,
+        overwrite=False,
+    ):
+        if image:
+            self.image = image
+        if base_image:
+            self.base_image = base_image
+        if commands:
+            self.with_commands(commands, overwrite=overwrite)
+        if requirements:
+            self.with_requirements(requirements, overwrite=overwrite)
+        if extra:
+            self.extra = extra
+        if secret is not None:
+            self.secret = secret
+        if source:
+            self.source = source
+        if load_source_on_run:
+            self.load_source_on_run = load_source_on_run
+        if with_mlrun is not None:
+            self.with_mlrun = with_mlrun
+        if auto_build:
+            self.auto_build = auto_build
+
+    def with_commands(
```

Comment:
this duplicates the logic from BaseRuntime.
I think we can call this method from the BaseRuntime `with_commands`:
```
    def with_commands(
        self,
        commands: List[str],
        overwrite: bool = False,
        verify_base_image: bool = False,
    ):
       self.spec.build.with_commends(command, overwrite)
       if verify_base_image:
            .....
```
same for `with_requirements`

Reply:
Yes! Thank you, I meant to but forgot to do that.